airflow/models/dag.py

```diff
@@ -770,6 +770,18 @@ def get_is_paused(self, session=NEW_SESSION) -> None:
         """Return a boolean indicating whether this DAG is paused."""
         return session.scalar(select(DagModel.is_paused).where(DagModel.dag_id == self.dag_id))
 
+    @property
+    @provide_session
+    def bundle_name(self, session=NEW_SESSION) -> str:
+        """Return the name of the bundle this DAG is in."""
+        return session.scalar(select(DagModel.bundle_name).where(DagModel.dag_id == self.dag_id))
```

Comment:
I don’t quite like this, it opens a new db connection to do one single db query for each property access. Too much hidden side effect for a simply attribute access IMO.

Reply:
The challenge is the bundle and version are only in the db. Would you be happier with a method so you could pass a session?

Reply:
I made that switch. If you still aren't happy with this, maybe I can defer and refactor more once parsing changes are merged. I'm using this to allow (along with some other changes) `dag.sync_to_db` to continue working, which fixes hundreds of tests in a single go.